runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -2202,3 +2202,72 @@ TR_J9SharedCacheServerVM::getClassFromNewArrayType(int32_t arrayType)
       return TR_J9ServerVM::getClassFromNewArrayType(arrayType);
    return NULL;
    }
+
+void *
+TR_J9SharedCacheServerVM::getJ2IThunk(char *signatureChars, uint32_t signatureLength, TR::Compilation *comp)
+   {
+   return (void *)findPersistentThunk(signatureChars, signatureLength);
+   }
+
+void *
+TR_J9SharedCacheServerVM::setJ2IThunk(char *signatureChars, uint32_t signatureLength, void *thunkptr, TR::Compilation *comp)
+   {
+   return persistThunk(signatureChars, signatureLength, (U_8*)thunkptr - 8 /* start of thunk */, *((U_32 *)((U_8*)thunkptr -8)) + 8 /* size of thunk */);
+   }
+
+void *
+TR_J9SharedCacheServerVM::persistJ2IThunk(void *thunk)
+   {
+   return persistThunk(((TR_J2IThunk *)thunk)->terseSignature(), strlen(((TR_J2IThunk *)thunk)->terseSignature()), (uint8_t*)thunk, ((TR_J2IThunk *)thunk)->totalSize());
```

Comment:
Same question here: does the think exist at the server at this point?

Reply:
Same as above.